src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -452,6 +471,84 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
+    modalRef.componentInstance.selectedStudents = affectedStudentModels;
+    modalRef.componentInstance.selectedInstructors = affectedInstructorModels;
+    modalRef.componentInstance.extensionTimestamp = submissionEndTimestamp;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.sessionEditFormModel.timeZone;
+
+    return new Observable((subscribeIsUserAccept) => {
+      modalRef.componentInstance.onConfirmExtensionCallBack.subscribe(() => {
+        this.removeDeadlines(affectedStudentModels, affectedInstructorModels);
+        modalRef.componentInstance.isSubmitting = false;
+        modalRef.close();
+        subscribeIsUserAccept.next(true);
+      }, () => {
+        subscribeIsUserAccept.next(false);
+      });
+    });
+  }
+
+  private getIndividualDeadlinesToDelete(submissionEndTimestamp: number): [
+    Record<string, number>, Record<string, number>,
+  ] {
+    const studentDeadlinesToDelete = DeadlineExtensionHelper.getDeadlinesBeforeOrEqualToEndTime(
+      this.studentDeadlines, submissionEndTimestamp);
+    const instructorDeadlinesToDelete = DeadlineExtensionHelper.getDeadlinesBeforeOrEqualToEndTime(
+      this.instructorDeadlines, submissionEndTimestamp);
+    return [studentDeadlinesToDelete, instructorDeadlinesToDelete];
+  }
+
+  /**
+   * Get models for individuals whose deadline extensions are before or equal to the new session end time.
+   */
+  private getAffectedIndividualModels(
+    submissionEndTimestamp: number,
+    affectedStudentDeadlines: Record<string, number>,
+    affectedInstructorDeadlines: Record<string, number>,
+  ): [StudentExtensionTableColumnModel[], InstructorExtensionTableColumnModel[]] {
+    const affectedStudents = this.studentsOfCourse.filter((student) => affectedStudentDeadlines[student.email]);
+    const affectedInstructors = this.instructorsOfCourse
+      .filter((instructor) => affectedInstructorDeadlines[instructor.email]);
+
+    const affectedStudentModels = DeadlineExtensionHelper.mapStudentsToStudentModels(
+      affectedStudents, affectedStudentDeadlines, submissionEndTimestamp,
+    );
+    const affectedInstructorModels = DeadlineExtensionHelper.mapInstructorsToInstructorModels(
+      affectedInstructors, affectedInstructorDeadlines, submissionEndTimestamp,
+    );
+    return [affectedStudentModels, affectedInstructorModels];
+  }
+
+  /**
+   * Update and set deadlines after filtering out extensions that exist before or equal to the new session end time.
+   */
+  private removeDeadlines(students: StudentExtensionTableColumnModel[],
+    instructors: InstructorExtensionTableColumnModel[],
+  ): void {
+    this.studentDeadlines = DeadlineExtensionHelper
+      .getUpdatedDeadlines(students, this.studentDeadlines, DeadlineHandlerType.DELETE);
+    this.instructorDeadlines = DeadlineExtensionHelper
+      .getUpdatedDeadlines(instructors, this.instructorDeadlines, DeadlineHandlerType.DELETE);
+  }
```

Comment:
Comment should be changed (probably can just remove entirely), because the method doesn't match the comment. If I understand this correctly, all it really does is remove the students and instructors that are passed in as arguments. I don't see anything about filtering out extensions.